narwhal/executor/src/notifier.rs

```diff
@@ -56,6 +56,15 @@ impl<State: ExecutionState + Send + Sync + 'static> Notifier<State> {
                     )
                     .await;
             }
+            // this is temporary
+            // we will get explicit signal from consensus on where is commit boundary in the future
+            if index.batch_index + 1
+                == index.consensus_output.certificate.header.payload.len() as u64
```

Comment:
Why does this indicate the commit boundary?

Reply:
See the comment above this line and/or PR description - it is not a commit boundary yet, however with this _some_ boundary so that we can start implementing checkpoints.

We will implement proper detection for commit boundary at some point later

Reply:
I see, so it is an arbitrary (but deterministic) moment that we use to decide when to checkpoint?

Reply:
Yes.

It is arbitrary in terms of safety(as long as everyone agree on what it is). Narwhal commit boundary is an optimal place for it in terms of performance (you essentially get a checkpoint after narwhal commit). Certificate boundary (like in this PR) is ok-ish, it is more frequent then necessarily, but still should work to begin with